src/sqlfe/general/Submission.java

```diff
@@ -176,60 +174,114 @@ public String toString() {
 				+ queryEvals + ", totalPoints=" + totalPoints + "]";
 	}
 	
+	// Skip comments or blank spaces before reaching a new question or an instructor comment
+	private String parseComments(String line, BufferedReader br , PrintWriter commWriter, boolean isNewQuestion ){
+		while (line != null && Utilities.isInstructorComment(line) && !isNewQuestion) {			
+		// skip remaining instructor comment lines with question text
+			line = Utilities.skipInstructorComments(br, line);
+			//System.out.println("line after skipping instructor comments with question text is: >" + line + "<");
+			
+			// skip any blank lines after instructor comments
+			line = Utilities.skipBlankLines(br, line);
+			//System.out.println("line after skipping blanks after instructor comments is: >" + line + "<");
+			
+			// process any user comments above the answer
+			line = Utilities.processUserComments(br, line, commWriter, submissionFileName);
+			//System.out.println("line after skipping user comments above answer is: >" + line + "<");
+			
+			// skip any remaining blank lines before answer
+			line = Utilities.skipBlankLines(br, line);
+			//System.out.println("line after skipping any remaining blank lines before answer is: >" + line + "<");
+		
+			// check the new line to see if is new question instructor comment
+			if (line != null && Utilities.isInstructorComment(line)) {
+		        if (Utilities.isQuestionFound(line)) {				// start of new question
+		        	isNewQuestion = true;
+		        	//System.out.println("parsing instructor comments, but found new question");
+		        } else {
+		        	//System.out.println("parsing instructor comments, found another instructor comment for same question");
+		        }
+			}
```

Comment:
Can we indent out nested if-blocks like this one? Or is this style carried over from Paul's code?

Reply:
We can, but before we do, we might need structural coverage on it which Jack could let us know through his tests

Reply:
Okay let's just plan to keep an eye on this then and we can change after merge + Jack's stuff